### ISSUE TITLE
The ANALYSE module analyzes more than just training set data.

### ISSUE THREAD

**Issue by Echo9573** [sql-machine-learning/sqlflow#825]
The current xgboost model automatically splits a validation set (30%) for verification during training, so the model actually is trained based on 70% of the train . Therefore, when using Shap, only 70% of the data should be interpreted. However, in the existing ANALYZE module statement, all training data was selected for analysis.

**Comment by weiguoz** [comment]
Good catch!
As discussed in the issues:
- [Add EVALUATE keyword](https://github.com/sql-machine-learning/sqlflow/issues/674)
- [Define the behavior of validation during training](https://github.com/sql-machine-learning/sqlflow/issues/675) 

We are going to avoid splitting the dataset implicitly. Then, the user can specify the same dataset between training and analyzer.

**Comment by tonyyang-svail** [comment]
@Echo9573 appreciated the great feedback. We will make sure this feature is on the top of our priority list for the next release.

**Comment by typhoonzero** [comment]
- Add EVALUATE keyword
- Define the behavior of validation during training

These features are already supported in the latest SQLFlow.
